Model/CreditCardPaymentMethod.php

```diff
@@ -0,0 +1,189 @@
+<?php
+/**
+ * Shop System Plugins:
+ * - Terms of Use can be found under:
+ * https://github.com/wirecard/oxid-ee/blob/master/_TERMS_OF_USE
+ * - License can be found under:
+ * https://github.com/wirecard/oxid-ee/blob/master/LICENSE
+ */
+
+namespace Wirecard\Oxid\Model;
+
+use OxidEsales\Eshop\Application\Model\Payment;
+use OxidEsales\Eshop\Core\Config;
+use OxidEsales\Eshop\Core\Registry;
+use Wirecard\Oxid\Core\Helper;
+
+use Wirecard\PaymentSdk\Config\Config as Wirecard_Config;
+use Wirecard\PaymentSdk\Config\CreditCardConfig;
+use Wirecard\PaymentSdk\Entity\Amount;
+use Wirecard\PaymentSdk\Transaction\CreditCardTransaction;
+use Wirecard\PaymentSdk\Transaction\Transaction;
+
+/**
+ * Class CreditCardPaymentMethod
+ *
+ */
+class CreditCardPaymentMethod extends Payment_Method
+{
+    /**
+     * @inheritdoc
+     */
+    protected static $_sName = 'creditcard';
+
+    /**
+     * @inheritdoc
+     *
+     * @return Transaction
+     */
+    public function getTransaction()
+    {
+        return new CreditCardTransaction();
+    }
+
+    /**
+     * @inheritdoc
+     *
+     * @param Payment $oPayment
+     *
+     * @return Wirecard_Config
+     */
+    public function getConfig($oPayment)
+    {
+        $oConfig = parent::getConfig($oPayment);
+
+        $oCreditCardConfig = new CreditCardConfig();
+
+        if (!empty($oPayment->oxpayments__wdoxidee_maid->value)) {
+            $oCreditCardConfig->setNonThreeDCredentials(
+                $oPayment->oxpayments__wdoxidee_maid->value,
+                $oPayment->oxpayments__wdoxidee_secret->value
+            );
+        }
+
+        if (!empty($oPayment->oxpayments__wdoxidee_three_d_maid->value)) {
+            $oCreditCardConfig->setThreeDCredentials(
+                $oPayment->oxpayments__wdoxidee_three_d_maid->value,
+                $oPayment->oxpayments__wdoxidee_three_d_secret->value
+            );
+        }
+
+        $this->_addThreeDLimits($oPayment, $oCreditCardConfig);
+
+        $oConfig->add($oCreditCardConfig);
+
+        $this->_oLogger->debug("CreditCardPaymentMethod#getConfig():" . print_r($oConfig, true));
```

Comment:
I think you don't need to put the method or class name here – the logger should have access to the stack anyway.

Reply:
removed, was just debugging output